pkg/util/features/features.go

```diff
@@ -0,0 +1,252 @@
+package features
+
+import (
+	"fmt"
+	"maps"
+
+	"github.com/loft-sh/admin-apis/pkg/licenseapi"
+	stripe "github.com/stripe/stripe-go/v81"
+	stripeclient "github.com/stripe/stripe-go/v81/client"
+)
+
+const (
+	metadataQueryFmt = "metadata['%s']:'%s'"
+)
+
+type SyncedFeature struct {
+	Feature *stripe.EntitlementsFeature
+	Product *stripe.Product
+}
+
+func EnsureFeatures(
+	stripeClient *stripeclient.API,
+	syncedFeatures map[string]SyncedFeature,
+	features []*licenseapi.Feature,
+	additionalAttachAllProducts []stripe.Product,
+	isLimit bool,
+) error {
+	err := EnsureStripeFeatures(stripeClient, syncedFeatures, features, isLimit)
+	if err != nil {
+		return err
+	}
+
+	if err = EnsureFeatureProducts(stripeClient, syncedFeatures); err != nil {
+		return err
+	}
+
+	if !isLimit {
+		if err = EnsureAttachAll(stripeClient, syncedFeatures, additionalAttachAllProducts); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func EnsureStripeFeatures(stripeClient *stripeclient.API, syncedFeatures map[string]SyncedFeature, features []*licenseapi.Feature, isLimit bool) error {
+	for _, f := range features {
+		extraMetadata := map[string]string{}
+		if isLimit {
+			extraMetadata[licenseapi.MetadataKeyFeatureIsLimit] = licenseapi.MetadataValueTrue
+			f.Name = licenseapi.LimitsPrefix + f.Name
+		}
+
+		if f.Status == string(licenseapi.FeatureStatusHidden) {
+			extraMetadata[licenseapi.MetadataKeyFeatureIsHidden] = licenseapi.MetadataValueTrue
+		}
+
+		err := EnsureFeatureExists(stripeClient, syncedFeatures, f.Name, f.DisplayName, extraMetadata)
+		if err != nil {
+			return err
+		}
+
+		if f.Preview {
+			extraMetadata[licenseapi.MetadataKeyFeatureIsPreview] = licenseapi.MetadataValueTrue
+			err = EnsureFeatureExists(stripeClient, syncedFeatures, f.Name+"-preview", f.DisplayName+" [Preview]", extraMetadata)
+			if err != nil {
+				return err
+			}
+		}
+
+		if isLimit {
+			extraMetadata[licenseapi.MetadataKeyFeatureLimitType] = licenseapi.MetadataKeyFeatureLimitTypeActive
+			err = EnsureFeatureExists(stripeClient, syncedFeatures, f.Name+"-active", f.DisplayName+" [Active]", extraMetadata)
+			if err != nil {
+				return err
+			}
```

Comment:
If both of these are true, won't the limit feature also end up with the preview metadata key?